swarm/storage/resource.go

```diff
@@ -533,20 +620,18 @@ func (self *ResourceHandler) Update(name string, data []byte) (Key, error) {
 // Closes the datastore.
 // Always call this at shutdown to avoid data corruption.
 func (self *ResourceHandler) Close() {
+	self.cancelFunc()
 	self.ChunkStore.Close()
 }
 
-func (self *ResourceHandler) getBlock() (uint64, error) {
-	// get the block height and convert to uint64
-	var currentblock string
-	err := self.rpcClient.Call(&currentblock, "eth_blockNumber")
+func (self *ResourceHandler) GetBlock() (uint64, error) {
+	ctx, cancel := context.WithCancel(self.ctx)
+	defer cancel()
```

Comment:
It doesn't really make sense to create a context with a deferred cancel which will only be called when the API method being called with the context returns.